app/src/main/java/org/thoughtcrime/securesms/notifications/MarkReadReceiver.kt

```diff
@@ -0,0 +1,153 @@
+                shortenExpiryOfDisappearingAfterRead(context, it)
+            }
+        }
+
+        private fun hashToDisappearAfterReadMessage(
+            context: Context,
+            markedReadMessages: List<MarkedMessageInfo>
+        ): Map<String, MarkedMessageInfo>? {
+            val loki = DatabaseComponent.get(context).lokiMessageDatabase()
+
+            return markedReadMessages
+                .filter { it.guessExpiryType() == ExpiryType.AFTER_READ }
+                .associateByNotNull { it.expirationInfo.run { loki.getMessageServerHash(id, isMms) } }
+                .takeIf { it.isNotEmpty() }
+        }
+
+        private fun shortenExpiryOfDisappearingAfterRead(
+            context: Context,
+            hashToMessage: Map<String, MarkedMessageInfo>
+        ) {
+            hashToMessage.entries
+                .groupBy(
+                    keySelector =  { it.value.expirationInfo.expiresIn },
+                    valueTransform = { it.key }
+                ).forEach { (expiresIn, hashes) ->
+                    SnodeAPI.alterTtl(
+                        messageHashes = hashes,
+                        newExpiry = nowWithOffset + expiresIn,
+                        publicKey = TextSecurePreferences.getLocalNumber(context)!!,
+                        shorten = true
+                    )
+                }
+        }
+
+        private fun sendReadReceipts(
+            context: Context,
+            markedReadMessages: List<MarkedMessageInfo>
+        ) {
+            if (isReadReceiptsEnabled(context)) {
+                markedReadMessages.map { it.syncMessageId }
+                    .filter { shouldSendReadReceipt(Recipient.from(context, it.address, false)) }
+                    .groupBy { it.address }
+                    .forEach { (address, messages) ->
+                        messages.map { it.timetamp }
+                            .let(::ReadReceipt)
+                            .apply { sentTimestamp = nowWithOffset }
+                            .let { send(it, address) }
+                    }
+            }
+        }
+
+        private fun fetchUpdatedExpiriesAndScheduleDeletion(
+            context: Context,
+            hashToMessage: Map<String, MarkedMessageInfo>
+        ) {
+            @Suppress("UNCHECKED_CAST")
+            val expiries = SnodeAPI.getExpiries(hashToMessage.keys.toList(), TextSecurePreferences.getLocalNumber(context)!!).get()["expiries"] as Map<String, Long>
+            hashToMessage.forEach { (hash, info) -> expiries[hash]?.let { scheduleDeletion(context, info.expirationInfo, it - info.expirationInfo.expireStarted) } }
+        }
+
+        private fun scheduleDeletion(
+            context: Context?,
+            expirationInfo: ExpirationInfo,
+            expiresIn: Long = expirationInfo.expiresIn
+        ) {
+            android.util.Log.d(TAG, "scheduleDeletion() called with: expirationInfo = $expirationInfo, expiresIn = $expiresIn")
```

Comment:
This one should probably be the libsession Log.d

Reply:
I might just delete it.